### ISSUE TITLE
Support LIKE operator on the client

### ISSUE THREAD

**Issue by rclai** [meteor/postgres-packages#36]
This will probably map to the Mongo `$regex` operator somehow.

https://github.com/meteor/postgres-packages/blob/6fd21592357cec9e439a306b118398f86a7d1e42/packages/bookshelf/knex.js#L762

Example:

``` js
Todos.where('content', 'like', '%Test%').fetch(); // => { content: { $regex: new RegExp('Test') } }
```

**Comment by stubailo** [comment]
Sounds cool!

**Comment by robfallows** [comment]
Are these the use cases (and is my use of greedy matching valid)?

```
Todos.where('content', 'like', '%Test%').fetch(); // => {content: {$regex: /Test/}}
Todos.where('content', 'like', 'Test%').fetch(); // => {content: {$regex: /^Test/}}
Todos.where('content', 'like', '%Test').fetch(); // => {content: {$regex: /Test$/}}
//... and more complex variations such as ...
Todos.where('content', 'like', 'Test%123%999').fetch(); // => {content: {$regex: /^Test.*123.*999$/}}
```

And how to handle a case-insensitive search (`/test/i`) ... or maybe case insensitivity should be the norm?

**Comment by rclai** [comment]
I'm not sure if Postgres handles LIKE the same way MySQL does, I have to check.

**Comment by stubailo** [comment]
Yeah we shouldn't accidentally code ourselves into a corner where one's client-side code depends on the specific database on the server. If MySQL and Postgres support `LIKE` differently I'd be in favor of just not having it at all on the client side. Right now it is just equivalent to doing an `_.filter` with a regex anyway.

**Comment by rclai** [comment]
At the moment, my workaround is to do this:

``` js
Todos= new PG.Table('todos');

const stuff = Todos.minimongo.find({ text: { $regex: /blah/ } });
```

**Comment by stubailo** [comment]
Yeah I guess I'm saying a Minimongo-agnostic way is:

``` js
const stuff = Todos.fetch().filter(row => /blah/.test(row.text));
```

**Comment by robfallows** [comment]
So, at the risk of being contentious - what is the end-game?
- Is it to make all databases look like (mini)mongo, or
- is it to provide a reasonably high fidelity interface to each database, or
- maybe a unifying One ORM To Rule Them All, or
- perhaps it's just a toe in the water to see if _something_ can be put together to show what _might_ be possible (but it may not ever get into production)?

If it's the first, I feel it's going to end up as a lowest common denominator, with only MongoDB being catered for at the highest fidelity. Sure, taking the agnostic approach will work as an alternative to LIKE, and it will probably work whatever the database (since it's minimongo at the end of the day). However, it isn't LIKE, and if I'm an experience SQL developer I'm going to get annoyed if I can't use SQL (or SQL-like) syntax. The same goes for NOT LIKE, IN, NOT IN etc.

If it's the second, then patching up 3rd party APIs to work in a way which permits us to use minimongo cursors, for example with Blaze, isn't going to cut the mustard.

The third approach would likely be so fundamentally breaking to the Meteor ecosystem that my brain hurts just thinking about it.

If it's the final option, then where's the harm in implementing a few things which may or may not see the light of day, but will give useful information to help formulate the true path forward?
